### PR TITLE
Sync with plugin-starter-template

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Downstream changes from https://github.com/mattermost/mattermost-plugin-starter-template

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
             "display_name": "Secret Message",
             "type": "custom",
             "help_text": "The message posted by the demo plugin when the secret phrase is detected.",
-            "default": "Yay! The random secret string was posted!\n\nGo to the settings page for this plugin in the system console to generate a new random secret."
+            "default": "Yay! The random secret string was posted! Go to the settings page for this plugin in the system console to generate a new random secret."
```
Comment (hanzei) [inline plugin.json]: I had to change this because `\n` did play nicely with the js world.

File: server/activate_hooks.go [hunk 2]
```diff
@@ -96,15 +94,5 @@ func (p *Plugin) OnDeactivate() error {
...
-	path, err := p.API.GetBundlePath()
-	if err != nil {
-		return false, errors.Wrap(err, "failed to get bundle path")
-	}
-
-	manifest, _, err := model.FindManifest(path)
-	if err != nil {
-		return false, errors.Wrap(err, "failed to find manifest file")
-	}
-
```
Comment (marianunez) [inline server/activate_hooks.go]: Is this intentionally deleted?

Comment (hanzei) [reply server/activate_hooks.go]: Yes. The method can reuse https://github.com/mattermost/mattermost-plugin-demo/pull/91/files#diff-3407a86002fad9e52eb18c8919aa27b5R11 and doesn't have to fetch the manifest itself

File: build/manifest/main.go [hunk 3]
```diff
@@ -10,19 +10,36 @@ import (
...
-
-var manifest = struct {
-	Id      string
-	Version string
-}{
-	Id:      "%s",
-	Version: "%s",
+const pluginIDGoFileTemplate = `// This file is automatically generated. Do not modify it manually.
+
+package main
```
Comment (jfrerich) [inline build/manifest/main.go]: @hanzei is was there some funky merging here?  package main defined more than once..

**Comment by hanzei** [comment]
Let's wait for https://github.com/mattermost/mattermost-plugin-starter-template/pull/89

File: build/manifest/main.go (hunk 3 above)
Comment (hanzei) [reply build/manifest/main.go]: I'm not sure I'm getting was the issue is. The generated file gets placed into `server`, which contains all needed `.go` files for the plugin to build. They are all in `package main`.

Comment (jfrerich) [reply build/manifest/main.go]: I was thrown off by the additional `package main` line while skimming through the files and didn't see that is included as a string.

**Comment by hanzei** [comment]
@DHaussermann Would you please give this PR a smoke test?

**Comment by hanzei** [comment]
Let's wait and cut `v0.6.0` first.

**Pull request merged**